[OVS Offload] Updated README and configs for back to back setup (#8)
ipdk-io/ipu-playbook: ovs_offload/README.md

# Linux Networking with OVS Offload

## Introduction

- The ovs_offload_lnw.py tool can be used on an host server connected with Intel® Infrastructure Processing Unit via PCIe.

> - It is supported on IPU SDK Release version >= 1.7.0 and makes use of the P4 package fxp-net_linux-networking.pkg

Reviewer: Random capitalization

```suggestion
- It is supported on IPU SDK release version >= 1.7.0 and makes use of the P4 package fxp-net_linux-networking.pkg
```

Reviewer 2: ```suggestion
- It is supported by IPU SDK Release version >= 1.7.0 and makes use of the P4 package fxp-net_linux-networking.pkg
```

- It creates the configuration to run the Linux Networking Recipe with OVS Offloaded to the IPU ACC.

> - It can be used to start infrap4d on the ACC and program the p4rt-ctl runtime rules to configure ACC Port representors for the Host IDPF interfaces and IPU Physical Port 0 and Port 1.

Reviewer: ```suggestion
- It can be used to start infrap4d on the ACC and program using the p4rt-ctl runtime rules to configure ACC port representors for the Host IDPF interfaces and IPU Physical Port 0 and Port 1.
```

Reviewer 2: ```suggestion
- It can be used to start infrap4d on the ACC and program the p4rt-ctl runtime rules to configure ACC Port representors for the Host IDPF interfaces and IPU Physical Ports 0 and 1.
```

- It can set up OVS bridges on the ACC using the port representors and configure VM namespaces for the IDPF interfaces on the Host.

> - Transport (Default mode): The tool generates the OVS bridge configuration for transport Mode with IPv4 encapsulation, use helper script 6_acc_ovs_bridge.sh.

Reviewer: ```suggestion
- Transport (Default mode): The tool generates the OVS bridge configuration for transport mode with IPv4 encapsulation, use helper script 6_acc_ovs_bridge.sh.
```

Reviewer 2: ```suggestion
- Transport (Default mode): The tool generates the OVS bridge configuration for transport Mode with IPv4 encapsulation. Use helper script 6_acc_ovs_bridge.sh.
```
Comma splice. The two clauses express different thoughts and have different structures. They should be separate sentences.

> - Tunnel (Optional mode): The tool also generates the OVS bridge configuration for Tunnel Mode with VXLAN encapsulation, use helper script acc_ovs_vxlan.sh. Refer VXLAN section.

Reviewer: ```suggestion
- Tunnel (Optional mode): The tool also generates the OVS bridge configuration for tunnel mode with VXLAN encapsulation, use helper script acc_ovs_vxlan.sh. Refer VXLAN section.
```

Reviewer 2: ```suggestion
- Tunnel (Optional mode): The tool also generates the OVS bridge configuration for Tunnel Mode with VXLAN encapsulation. Use helper script acc_ovs_vxlan.sh. See the VXLAN section.
```

> ### Supports two Setup Topologies

Reviewer: ```suggestion
### Supports Two Setup Topologies
```

We can either do a `Support Two Setup Topologies` with capitalization of first letter of all words (commonly used and we have used it in this and other documentation) or simply use `Support two setup topologies` -- but we should stay consistent

#### All in one configuration

> - IPU adapter and a Link Partner NIC is connected to a single Host server. Use Default ipu-playbook/ovs_offload/config.yaml

Reviewer 2: ```suggestion
- IPU adapter and a Link Partner NIC is connected to a single Host server. Use default ipu-playbook/ovs_offload/config.yaml.
```

- Host1 IPU Port0 <-----> Host1 Link Partner Port0
- Host1 IPU Port1 <-----> Host1 Link Partner Port1

> #### Two IPU Host servers connected back to back

Reviewer 2: ```suggestion
#### Two IPU Host servers connected back-to-back
```

> - 2 IPU Adapters are connected to two Host servers via PCIe and the IPU Physical Ports on the two servers are connected back to back.

Reviewer 2: ```suggestion
- Two IPU Adapters are connected to two Host servers via PCIe, and the IPU Physical Ports on the two servers are connected back-to-back.
```

- Host1 IPU Port0 <-----> Host2 IPU Port0
- Host1 IPU Port1 <-----> Host2 IPU Port1

> - For IPU Host 1 server copy config_host1.yaml to config.yaml.

Reviewer 2: ```suggestion
- For IPU Host 1 server, copy config_host1.yaml to config.yaml.
```

> - For IPU Host 2 server copy config_host2.yaml to config.yaml.

Reviewer 2: ```suggestion
- For IPU Host 2 server, copy config_host2.yaml to config.yaml.
```

            [-S socket-path] [-T features] [command [flags]]
```

## Test Configuration

- The tool uses the config.yaml file to program the rules for OVS Offload on the IPU Adapter and configure the IDPF interfaces on the Host server.

> - The tool can be used in All in one configuration or 2 IPU Servers connected Back to back.

Reviewer 2: ```suggestion
- The tool can be used in an all-in-one configuration or as two IPU Servers connected back-to-back.
```

- More information on the configuration below

> ### All in one Test Configuration (config.yaml)

Reviewer 2: ```suggestion
### All-in-one configuration (config.yaml)
```

> - All in one configuration IPU adapter and a Link Partner NIC is connected to a single Host server. Use Default ipu-playbook/ovs_offload/config.yaml

Reviewer 2: ```suggestion
- All-in-one configuration. IPU adapter and a Link Partner NIC are connected to a single Host server. Use default ipu-playbook/ovs_offload/config.yaml.
```

> - Update the file: config.yaml for the specific test setup. Change the management IP, username, and password for imc and acc if they are different.

Reviewer 2: ```suggestion
- Update the file `config.yaml` for the specific test setup. Change the management IP, username, and password for imc and acc if they are different.
```

> - Update the test_params section as required for the setup with the correct host, imc and acc script paths.

Reviewer 2: ```suggestion
- Update the test_params section as required for the setup with the correct host, imc, and acc script paths.
```

> - Update the idpf_interface, vf_interfaces in the config if the interface names are different.

Reviewer 2: ```suggestion
- Update the idpf_interface and vf_interfaces in the config if the interface names are different.
```

    remote_br_tun_ip: ['1.1.1.2','2.1.1.2']
```

> ### 2 IPU Servers connected back to back configuration (config_host1.yaml and config_host2.yaml)

Reviewer 2: ```suggestion
### Back-to-back configuration (config_host1.yaml and config_host2.yaml)
```

1. Section names should be short and to the point.
2. The section names should have parallel construction and should emphasize their distinction (_all-in-one_ vs. _back-to-back_).
This makes the structure of the document easier to understand.

> - 2 IPU Host servers are connected back to back.

Reviewer 2: ```suggestion
- Two IPU Host servers are connected back-to-back.
```

- Host1 IPU Port0 <-----> Host2 IPU Port0
- Host1 IPU Port1 <-----> Host2 IPU Port1

> - For IPU Host 1 server copy config_host1.yaml to config.yaml.

Reviewer 2: ```suggestion
- For IPU Host 1 server, copy config_host1.yaml to config.yaml.
```

> - For IPU Host 2 server copy config_host2.yaml to config.yaml.

Reviewer 2: ```suggestion
- For IPU Host 2 server, copy config_host2.yaml to config.yaml.
```

> - Run ovs_offload_lnw.py in the 2 IPU Host servers after copying the respective configs as above.

Reviewer 2: ```suggestion
- Run ovs_offload_lnw.py in the two IPU Host servers after copying the respective configs as above.
```

> - Update the file: config.yaml for the specific test setup. Change the management IP, username, and password for imc and acc if they are different.

Reviewer 2: ```suggestion
- Update the config.yaml file for the specific test setup. Change the management IP, username, and password for imc and acc if they are different.
```

> - Update the test_params section as required for the setup with the correct host, imc and acc script paths.

Reviewer 2: ```suggestion
- Update the test_params section as required for the setup, with the correct host, imc, and acc script paths.
```

> - Update the idpf_interface, vf_interfaces name in the config if the interface names are different.

Reviewer 2: ```suggestion
- Update the idpf_interface and vf_interface names in the config if the interface names are different.
```

> - Note that the lp_interfaces and lp_interface_ip fields below is empty as we are running in back to back mode and there is no Link Partner NIC connected to the same server.

Reviewer 2: ```suggestion
- Note that the lp_interfaces and lp_interface_ip fields below are empty, as we are running in back-to-back mode and there is no Link Partner NIC connected to the same server.
```

    - Configure TMUX session - test2_p4rt delete the p4rt-ctl rules and delete the OVS bridges
    - Configure TMUX session - test1_infrap4d, login to ACC and stop infrap4d,

## Run the script on the IPU Host

> - Run below commands in the IPU Host server to configure OVS Offload on the IPU ACC and configure the Host IDPF Interfaces.

Reviewer 2: ```suggestion
- Run the following commands on the IPU Host server to configure OVS Offload on the IPU ACC and to configure the Host IDPF Interfaces.
```

[root@ipu-acc ~]# kill 11188
```

### Appendix

> - For IPU SDK version < 1.6.1 use below scripts

Reviewer 2: ```suggestion
- For IPU SDK version < 1.6.1 use the following scripts
```

> ### ovs_offload_lnw_v3.py : (P4:fxp-net_linux-networking_v3.p4, IPU SDK Release 1.6.0, 1.6.1)

Reviewer 2: ```suggestion
### ovs_offload/ovs_offload_lnw_v3.py
```

> 1. This is a python script : **ovs_offload/ovs_offload_lnw_v3.py** that can be used with **P4: fxp-net_linux-networking_v3.p4** for release 1.6.0,1.6.1

Reviewer 2: ```suggestion
This Python script can be used with `P4: fxp-net_linux-networking_v3.p4` for release 1.6.0 and 1.6.1.
```
